packages/hardhat-ethers/src/helpers.ts

```diff
@@ -24,35 +46,172 @@ export function getContractFactory(
 export async function getContractFactory(
   hre: HardhatRuntimeEnvironment,
   nameOrAbi: string | any[],
-  bytecodeOrSigner?: ethers.Signer | ethers.utils.BytesLike | string,
+  bytecodeOrFactoryOptions?:
+    | (ethers.Signer | FactoryOptions)
+    | (ethers.utils.BytesLike | string),
   signer?: ethers.Signer
 ) {
   if (typeof nameOrAbi === "string") {
-    return getContractFactoryByName(
+    const contractFactory = await getContractFactoryByName(
       hre,
       nameOrAbi,
-      bytecodeOrSigner as ethers.Signer | undefined
+      bytecodeOrFactoryOptions as ethers.Signer | FactoryOptions | undefined
     );
+
+    if (contractFactory.bytecode === "0x") {
+      throw new NomicLabsHardhatPluginError(
+        pluginName,
+        `The requested contract, ${nameOrAbi}, is an abstract contract.
```

Comment:
`You are trying to create a contract factory for the contract ${nameOrAbi}, which is abstract and can't be deployed.`

Reply:
In general, when a message like this has to be defined, you can discuss it in the `#hardhat-copy` channel.